File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        cur = list1
        index = 0
        while index!=b:
            if index+1 == a:
                start_node = cur
            cur = cur.next
            index+=1
        end_node = cur.next
        
        cur = list2
        while cur.next!=None:
            cur = cur.next
        
        start_node.next = list2
        cur.next = end_node
        
        return list1

File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeInBetween(list1 *ListNode, a int, b int, list2 *ListNode) *ListNode {
    var cur *ListNode = list1
    var cur2 *ListNode = list2
    var index int = 0
    var startNode *ListNode = nil
    var endNode *ListNode = nil
    
    for index!=b{
        if index+1 == a{
            startNode = cur
        }
        cur = cur.Next
        index++
    }
    endNode = cur.Next
    
    
    for cur2.Next!=nil{
        cur2 = cur2.Next
    }
    
    startNode.Next = list2
    cur2.Next = endNode
    
    return list1
    
}